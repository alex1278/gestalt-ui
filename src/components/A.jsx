import styled, { withTheme, css } from 'styled-components';

const AStyle = styled.a`
  font-size: 12px;
  text-decoration: none;
  ${props => props.bubble && css`
    background-color: ${props.theme.colors['$md-grey-200']};
    border-radius: 15px;
    padding-left: 4px;
    padding-right: 4px;
    font-weight: 400;
    color: ${props.theme.colors['$md-blue-900']};;
  `};
  ${props => props.block && 'display: block'};
`;

export default withTheme(AStyle);
