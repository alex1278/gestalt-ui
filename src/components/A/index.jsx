import styled from 'styled-components';

const EnhancedA = styled.a`
  text-decoration: none;
  font-size: 12px;
`;

export default EnhancedA;
