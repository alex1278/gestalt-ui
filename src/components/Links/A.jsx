import styled, { css } from 'styled-components';

const AStyle = styled.a`
  font-size: 13px;
  text-decoration: none;
  ${props => props.bubble && css`
    background-color: ${props.theme.colors['$md-grey-200']};
    border-radius: 15px;
    padding-left: 4px;
    padding-right: 4px;
    font-weight: 400;
  `};
  ${props => props.block && 'display: block'};
  ${props => props.primary && `color: ${props.theme.colors['$md-blue-600']}`};
`;

export default AStyle;
