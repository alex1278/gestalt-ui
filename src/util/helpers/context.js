/**
 * generateContextEntityState
 * @param {*} params
 */
export function generateContextEntityState(params) {
  const entity = {
    id: params.environmentId || params.workspaceId || null,
    key: null,
  };

  if (params.workspaceId && params.environmentId) {
    entity.key = 'environments';
  }

  if (params.workspaceId && !params.environmentId) {
    entity.key = 'workspaces';
  }

  // if (params.workspaceId && params.environmentId && params.providerId) {
  //   entity.id = params.providerId;
  //   entity.key = 'providers';
  // }

  return entity;
}
