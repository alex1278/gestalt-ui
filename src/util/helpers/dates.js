import moment from 'moment';

export const getNow = () => moment();

export const getAgo = (num = 0, unit = 'day') =>
  moment().subtract(num, unit);

export const isRecentWithin = (timestamp, num, unit) =>
  moment(timestamp).isBetween(getAgo(num, unit), getNow());
