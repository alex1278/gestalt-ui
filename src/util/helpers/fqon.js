// sad hack for parsing url

export default function fqonParent(organization) {
  if (organization.org) {
    const urlParser = document.createElement('a');
    urlParser.href = organization.org.href;
    return urlParser.pathname.replace('/', '').split('/')[0];
  }

  return organization;
}
