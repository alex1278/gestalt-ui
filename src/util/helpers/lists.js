export function insertItem(array, payload) {
  return [...array, payload];
}

export function removeItem(array, payload) {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a === payload), 1);

  return newArray;
}

export function removeItemById(array, id, keyField = 'id') {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a[keyField] === id), 1);

  return newArray;
}

export function updateItem(array, payload) {
  return array.map((item) => {
    if (item.id === payload.id) {
      return {
        ...payload,
      };
    }

    // Leave every other item unchanged
    return item;
  });
}
