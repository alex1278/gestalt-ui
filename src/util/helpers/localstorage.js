/**
 * getItem
 * @param {*} key
 */
export const getItem = (key) => {
  try {
    return localStorage.getItem(key);
  } catch (err) {
    return undefined;
  }
};

/**
 * saveItem
 * @param {*} key
 * @param {*} value
 */
export const saveItem = (key, value) => {
  try {
    localStorage.setItem(key, value);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};

/**
 * removeItem
 * @param {*} key
 * @param {*} value
 */
export const removeItem = (key) => {
  try {
    localStorage.removeItem(key);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};
