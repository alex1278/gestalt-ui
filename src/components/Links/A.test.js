import React from 'react';
import A from './A';
import { renderWithTheme } from '../../../test/helpers';

describe('(Component) A', () => {
  it('mounts without additional props', () => {
    const { container } = renderWithTheme(<A href="http://wassup" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('mounts the bubble prop', () => {
    const { container } = renderWithTheme(<A href="http://wassup" bubble />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
