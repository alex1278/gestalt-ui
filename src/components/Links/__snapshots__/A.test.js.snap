// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) A mounts the bubble prop 1`] = `
.c0 {
  font-size: 13px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #eee;
  border-radius: 15px;
  padding-left: 4px;
  padding-right: 4px;
  font-weight: 400;
}

<a
  class="c0"
  href="http://wassup"
/>
`;

exports[`(Component) A mounts without additional props 1`] = `
.c0 {
  font-size: 13px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="http://wassup"
/>
`;
